### ISSUE TITLE
Server only modules are being accessed in client buncle when using Cloud Storage plugin for S3

### ISSUE THREAD

**Issue by jgb-solutions** [payloadcms/next-payload#23]
I'm trying to use the cloud storage plugin. Unfortunately I can't because when I try to use the admin panel I get this:
`Module not found: Can't resolve 'fs'` 

<img width="1049" alt="Screenshot 2023-04-25 at 8 30 03 PM" src="https://user-images.githubusercontent.com/8051752/234436517-8afd881d-f031-4108-a59c-23cd0759af4b.png">

It's trying to load the cloudStorage plugin's dependencies. Especially 'fs'. I read `https://payloadcms.com/docs/admin/webpack` for webpack. But I'm using Next.js, the integration should be flawless I guess. How to fix this?

Also I'm using Zod to parse the `.env` variables for some config values for the plugin and also for generating types. Since the config is ran on both the server and the client Zod is erroring out because it can't read the env vars. I had to import dotenv to make the generation of types' script work, but can't do anything when it's trying to load the code in the admin panel.

**Comment by jmikrut** [comment]
Hey @jgb-solutions — you might have an issue related to the following issue which is open on the actual cloud storage plugin:

https://github.com/payloadcms/plugin-cloud-storage/issues/49

Can you check that and make sure that you are properly initializing the cloud storage plugin? 

TL;DR: the plugin is responsible _**itself**_ for adjusting your Webpack config, but if you don't actually initialize it at all, it won't adjust your Webpack config at all. You need to make sure that it is indeed executing otherwise you will see dependency issues like what you've reported.

This is indeed functional though. Take a look at how the `next-payload-demo` repo uses it:
https://github.com/payloadcms/next-payload-demo/blob/main/payload/payload.config.ts

I will close this but I am happy to help further - so just let me know if you can't get it to work. Might need to have you share your Payload config for further help though. 👍

**Comment by jgb-solutions** [comment]
@jmikrut thanks, it worked when I put everything together.

**Comment by IAmNatch** [comment]
@jgb-solutions Would be able to share the config you landed on? I'm still hitting this without avail.

**Comment by revnelson** [comment]
I'm using this plugin and it seems to be working, but when I query a media item the url is `/media/image_file_name.png` and there is no route handler for that so the image won't load. How can I get this to work?
